### ISSUE TITLE
`:VimwikiGenerateLinks [pattern]` and `auto_generate_links  = 1` when saving file

### ISSUE THREAD

**Issue by beuerle** [vimwiki/vimwiki#983]
There is an issue how vimwiki behaves when `auto_generate_links = 1` is set in the config. I'm not really sure how to resolve this.

When one generates specific links, for example in a subdirectory, with:

```vim
:VimwikiGenerateLinks subdir/*.md
```

vimwiki correctly only generates a list of all files in `subdir`:

```markdown
# Generated Links

- [note1](subdir/note1.md)
- [note2](subdir/note2.md)
[...]
```

However, upon saving the file (with `auto_generate_links = 1`) vimwiki goes wild and generates a list of **all** files/links in the wiki (because the info of only create the list for `subdir` is nowhere stored).

So if you generate customized link lists, you can't use `auto_generate_links = 1`.

A solution would be to put the used pattern into the header, after `Generated Links`?

**Comment by tinmarino** [comment]
> A solution would be to put the used pattern into the header, after Generated Links?

Yes using a placeholder like `  let g:vimwiki_links_header = 'Files in this subpage at __Dir__'`.

**Comment by tinmarino** [comment]
Ah and I vote in favor of the feature. As I am the only voter now => please go ahead.

**Comment by zamlz** [comment]
I too vote for this feature :+1:

**Comment by zamlz** [comment]
Might I suggest that it be something much cleaner? I'd like to have my headers be named whatever

```markdown
### Research Papers Published in 2019
<!-- Generated links for [pubs/*_2019.md] -->
```

This is in markdown, but you could do the samething in the other syntaxes. Basically the pattern definition could be put into the comment? The reason I'd personally prefer this is that when its built into HTML or as PDF, the programmatic description of how the links were generation is hidden now.

**Comment by ryanbrate** [comment]
+1 from me, but....is there any chance this could be extended to auto_generate_tags behaviour too?
I use :Diary-projectname: tags to enable linking of diary notes to specific projects, for easy search. 
It'd be awesome if I could maintain two separate tag lists: one for tags with pattern /Diary-.*/ and one for all non Diary tags.

**Comment by maximilianigl** [comment]
+1

**Comment by Whyglobaleyes** [comment]
Hi all,
I know it;s been a while but can I ask if this was ever implemented please?

**Comment by Whyglobaleyes** [comment]
@beuerle if this is still a problem then I've got a work-around that I think might help 
https://github.com/vimwiki/vimwiki/issues/803
